src/overlays/actors/ovl_En_Ot/z_en_ot.c

```diff
@@ -41,128 +81,1099 @@ const ActorInit En_Ot_InitVars = {
+
+    sp50.x = arg1->unk_20.x - temp_s1->world.pos.x;
+    sp50.y = arg1->unk_20.y - temp_s1->world.pos.y;
+    sp50.z = arg1->unk_20.z - temp_s1->world.pos.z;
+
+    sp44.x = arg1->unk_20.x - arg1->unk_38.x;
+    sp44.y = arg1->unk_20.y - arg1->unk_38.y;
+    sp44.z = arg1->unk_20.z - arg1->unk_38.z;
+
+    temp = Math3D_Parallel(&sp50, &sp44);
+    if ((arg1->unk_4C < SQ(temp_s1->speedXZ)) || (temp <= 0.0f)) {
+        ret = true;
+    } else {
+        temp = SQ(temp_s1->speedXZ) / arg1->unk_50;
+        sp2C = ABS(arg1->unk_54 - temp_s1->world.rot.x);
+        sp2C = (s32)(sp2C * temp) + 0xAAA;
+
+        sp34 = ABS(arg1->unk_56 - temp_s1->world.rot.y);
+
+        Math_SmoothStepToS(&temp_s1->world.rot.x, arg1->unk_54, 1, sp2C, 0);
+        sp2C = (s32)(sp34 * temp) + 0xAAA;
+        Math_SmoothStepToS(&temp_s1->world.rot.y, arg1->unk_56, 1, sp2C, 0);
+        Math_SmoothStepToS(&temp_s1->shape.rot.y, temp_s1->world.rot.y, 2, sp2C, 0);
+    }
+
+    return ret;
+}
+#else
+s32 func_80B5D470(GlobalContext* globalCtx, EnDno_ActorUnkStruct* arg1);
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ot/func_80B5D470.s")
+#endif
+
+void func_80B5D648(EnOt* this, GlobalContext* globalCtx) {
+    func_80B5B2E0(globalCtx, &this->actor.world.pos, this->unk_346, &this->unk_348, &this->unk_340);
+    Math_Vec3f_Copy(&this->unk_330, &this->actor.world.pos);
+    func_8013DCE0(globalCtx, &this->unk_330, &this->actor, &this->unk_2C0, globalCtx->setupPathList, this->unk_346, 0,
+                  0, this->unk_340, 0);
+    this->unk_32C = 0;
+    this->unk_2C0.unk_2C.x = 0.0f;
+    this->unk_2C0.unk_2C.y = 0.0f;
+    this->unk_2C0.unk_2C.z = 0.0f;
+    this->actor.gravity = 0.0f;
+    this->actor.speedXZ = 0.0f;
+    func_8013E1C8(&this->skelAnime, sAnimations, 1, &this->animIdx);
+    this->actor.flags |= 0x8000000;
+    this->actor.flags &= ~(0x8 | 0x1);
+    Actor_SetSwitchFlag(globalCtx, ENOT_GET_3F80(&this->actor));
+    this->actionFunc = func_80B5D750;
+}
+
+void func_80B5D750(EnOt* this, GlobalContext* globalCtx) {
+    if (!(this->unk_32C & 1) && !(this->unk_32C & 2)) {
+        func_8013DE04(globalCtx, &this->unk_2C0, func_8013DF3C, func_80B5D470, func_80B5D37C, func_8013E0A4);
+    }
+
+    Math_Vec3f_Copy(&this->actor.world.pos, &this->unk_330);
+
+    if (this->unk_2C0.unk_1C & 0x40) {
+        this->unk_32C |= 2;
+    }
+
+    if (this->unk_2C0.unk_1C & 0x80) {
+        this->unk_32C |= 1;
+    }
+
+    if ((this->unk_32C & 2) && (this->actor.xyzDistToPlayerSq <= 14400.0f)) {
+        this->unk_32C &= ~2;
+    }
+
+    if ((this->unk_32C & 1) && (this->actor.xzDistToPlayer <= 180.0f)) {
+        this->actor.flags &= ~0x8000000;
+        this->actor.flags |= (0x8 | 0x1);
+        if (D_80B5E884 != 0) {
+            func_80B5C9A8(this, globalCtx);
+        } else {
+            func_80B5CBA0(this, globalCtx);
+        }
+    }
+}
+
+void EnOt_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnOt* this = THIS;
+    s32 pad;
```

Comment:
```suggestion
    s32 pad;
    EnOt* this = THIS;
```
if it matches.

Reply:
Do we have to do this? I think it looks really bad.